modules/eks/main.tf

```diff
@@ -0,0 +1,29 @@
+resource "aws_cloudwatch_log_group" "main" {
+  name              = "/aws/eks/${var.cluster_name}/cluster"
+  retention_in_days = 14
+}
+
+resource "aws_eks_cluster" "eks_cluster" {
+  name     = var.cluster_name
+  role_arn = aws_iam_role.main.arn
+
+  enabled_cluster_log_types = [
+    "api",
+    "audit",
+    "authenticator",
+    "controllerManager",
+    "scheduler",
+  ]
+
+  vpc_config {
+    endpoint_public_access  = false
```

Comment:
This means we will need to specify an ingress? How will we route traffic to the cluster?

Reply:
Nop, this is for the API access. This means we'll need a bastion server to access the API. It's a security measurement (not expose the kubernetes API over internet)